### ISSUE TITLE
Add new type of plot: violinPlot

### ISSUE THREAD

**Issue by sylwiabr** [luna/Dataframes#25]

**Comment by Magalame** [comment]
I think I'd like to try that one out!

**Comment by mwu-tow** [comment]
Hi @Magalame — this task is all yours, if you have any questions or issues, don't hesitate to ask.

Some points of interest for you:
* https://github.com/luna/dataframes/blob/master/native_libs/third-party/matplotlib-cpp/matplotlibcpp.h — here we wrap the python code (we rewrite, as needed, matplotlib-cpp to use pybind11 library) into C++
* https://github.com/luna/dataframes/blob/master/native_libs/plotter/Matplotlib/Plot.cpp — exposing C++ as C API that Luna can call
* https://github.com/luna/dataframes/blob/master/src/Plot/Matplotlib.luna — Luna wrappers

**Comment by Magalame** [comment]
Thanks! I have to admit I don't have much C++ experience but I'll try just imitating what is already written

**Comment by Magalame** [comment]
I was wondering, what is the plan regarding the api more exactly? The `violin` plots? like for `kde` and `kde2`?

**Comment by mwu-tow** [comment]
@Magalame 
Yes, I'd expect something like this. Basically after completing C++ part, you should in Luna:
* add wrapper to exported C API method to `Matplotlib` class (in the `Matplotlib` module)
* add class `Violin` to the `Matplotlib` module, that contains all the parameters we allow to be passed from Luna to Seaborn
* add `violin` method to `Table` and `ChartBuilder` classes

To focus attention, you might want to start with recreating some example from seaborn documentation page and then build upon this. Or you can create brand new show-case. It's up to you!

**Comment by Magalame** [comment]
@mwu-tow sorry, I think I'll drop this one, C++ isn't my forte and Luna Studio consumes a fair amount of RAM on linux, it quickly get hard to work with it :/
